src/app/api/quickbooks/invoice/invoice.service.ts

```diff
@@ -517,15 +577,15 @@ export class InvoiceService extends BaseService {
     }
 
     // 2. if not, create payment in QB, sync payment in payment sync table and change invoice status to paid
-    if (!invoiceSync?.clientId) {
+    if (!invoiceSync?.recipientId) {
```

Comment:
```suggestion
    if (!invoiceSync.recipientId) {
```
invoiceSync is guaranteed to be present, since we have a check for it above